pkg/archive/parser_test.go

```diff
@@ -110,11 +111,161 @@ func TestParse(t *testing.T) {
 
 			res, err := p.Parse(tc.dir)
 			if tc.wantErr != nil {
-				assert.Equal(t, err.Error(), tc.wantErr.Error())
+				assert.Equal(t, tc.wantErr.Error(), err.Error())
 			} else {
 				assert.Nil(t, err)
 				assert.Equal(t, tc.want, res)
 			}
 		})
 	}
 }
+
+func TestParseGroupVersions(t *testing.T) {
+	tests := []struct {
+		name      string
+		files     []string
+		backupDir string
+		wantErr   error
+		want      map[string]metav1.APIGroup
+	}{
+		{
+			name:      "when there is no top-level resources directory, an error is returned",
+			backupDir: "/var/folders/zj/vc4ln5h14djg9svz7x_t1d0r0000gq/T/620385697",
+			wantErr:   errors.New("\"/var/folders/zj/vc4ln5h14djg9svz7x_t1d0r0000gq/T/620385697/resources\" not found"),
+		},
+		{
+			name:      "when there are no directories under the resources directory, an empty map is returned",
+			backupDir: "/var/folders/zj/vc4ln5h14djg9svz7x_t1d0r0000gq/T/620385697",
+			files:     []string{"/var/folders/zj/vc4ln5h14djg9svz7x_t1d0r0000gq/T/620385697/resources/"},
+			want:      map[string]metav1.APIGroup{},
+		},
+		{
+			name:      "when there is a mix of cluster-scoped and namespaced items for resources with preferred or multiple API groups, all group versions are correctly returned",
+			backupDir: "/var/folders/zj/vc4ln5h14djg9svz7x_t1d0r0000gq/T/620385697",
+			files: []string{
+				"/var/folders/zj/vc4ln5h14djg9svz7x_t1d0r0000gq/T/620385697/resources/clusterroles.rbac.authorization.k8s.io/v1-preferredversion/cluster/system/controller/attachdetach-controller.json",
```

Comment:
I appreciate your careful review because you're right. I missed the cluster directory at the same level as the preferred version directory. I'll add that line.